src/System.Runtime.Extensions/tests/System/Environment.MachineName.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Tests
+{
+    public class Environment_MachineName
+    {
+        // [Fact] Skip test for the mscolib changes to reach corefx
+        public void TestMachineNameProperty()
+        {
+            string computerName = GetComputerName();
+            Assert.Equal(computerName, Environment.MachineName);
+        }
+
+        [DllImport("kernel32.dll", EntryPoint = "GetComputerNameW")]
+        private unsafe static extern int GetComputerName(char* nameBuffer, ref int bufferSize);
+
+        [DllImport("libc")]
```

Comment:
Yes, calls to gethostname should use https://github.com/dotnet/corefx/blob/e0a215f3f959407be801c9e7837c9bb4d9df9fb6/src/Common/src/Interop/Unix/System.Native/Interop.GetHostName.cs